src/Core/Resolvers/SqlPaginationUtil.cs

```diff
@@ -322,10 +325,10 @@ RuntimeConfigProvider runtimeConfigProvider
                     // Since we are looking for primary keys we expect these columns to
                     // exist.
                     string safePK = GetExposedColumnName(entityName, pk, sqlMetadataProvider);
-                    if (!afterDict.ContainsKey(safePK))
+                    if (!exposedFieldNameToBackingColumn.ContainsKey(safePK))
```

Comment:
after further reflection, the name `safePK` could be `exposedFieldName` to be consistent with other parts of our codebase.